### PR TITLE
Release v.3.13.0

### PR DESCRIPTION
**Pull request by studioswong**
## Description

This PR is to release version 3.13.0 for the following major feature releases along with bug fixes:
- Implement new tree list with modular pipelines and search UI for new sidebar navigation. (#479)
- Implement element filters and further design updates to the filter panel for new sidebar navigation. (#454)
- Implement kedro telemetry in production to enable heap analysis for Kedro-viz. (#481, #487)
- Show decorated function's source code on code panel. (#493)
- Enable the display of entire parameter object with react-json-viewer on the metadata panel. (#494)

Bug fixes:
- Upgrade dev dependencies to solve snyk errors. (#486)
- Remove the old dagre graphing logic and the 'oldgraph' flag. (#486)
- Delete 'modularpipeline' flag. (#495)
- Fix run command suggestion. (#497)

## Development notes

This PR only contains updates to the release notes. 

## QA notes

Appreciate any comments on the release notes. 

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [x] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

## Legal notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

**Review by limdauto — APPROVED** [review]
AMAZING. LETS GOOO!

File: RELEASE.md [hunk 1]
```diff
@@ -8,6 +8,23 @@ Please follow the established format:
 - Include the ID number for the related PR (or PRs) in parentheses
 -->
 
+# Release 3.13.0
+
+## Major features and improvements
```
Comment (limdauto) [inline RELEASE.md:13]: We should start populating this as we go from now on

File: RELEASE.md [hunk 2]
```diff
@@ -8,6 +8,23 @@ Please follow the established format:
...
+
+- Implement new tree list with modular pipelines and search UI for new sidebar navigation. (#479)
+- Implement element filters and further design updates to the filter panel for new sidebar navigation. (#454)
+- Implement kedro telemetry in production to enable heap analysis for Kedro-viz. (#481, #487)
```
Comment (yetudada) [inline RELEASE.md:17]:
```suggestion
- Implement `kedro-telemetry` in production to enable heap analysis for Kedro-Viz. (#481, #487)
```

**Review by yetudada — APPROVED** [review]
🏆  I left a minor comment but otherwise we're all good to go!

**Pull request merged**